src/bodse/teams.py

```diff
@@ -0,0 +1,223 @@
+                    " be posted to MS Teams channel",
+                    RuntimeWarning,
+                )
+            else:
+                raise ModuleNotFoundError("pymsteams")
+
+        else:
+            LOG.info("Setup teams posts pointing to %s", self.webhook_url.host)
+            self.post(
+                f"{self.tool_name} - Startup",
+                "Setup posts to be sent to this channel for any errors or outputs.",
+            )
+
+    @pydantic.validator("webhook_url")
+    def validate_webhook_host(  # pylint: disable=no-self-argument
+        cls, value: pydantic.HttpUrl
+    ) -> pydantic.HttpUrl:
+        """Validate URL points to office webhook."""
+        end = "webhook.office.com"
+
+        if value.host is None:
+            raise ValueError("webhook URL invalid, no host given")
+
+        if not value.host.lower().endswith(end):
+            raise ValueError(
+                "webhook URL has incorrect host, expected host"
+                f" to end with '{end}' but got '{value.host}'"
+            )
+
+        return value
+
+    @functools.cached_property
+    def _tool_details_card(self) -> "pymsteams.cardsection":
+        """Create cardsection containing tool information."""
+        assert pymsteams is not None, "module 'pymsteams' not loaded"
+        section = pymsteams.cardsection()
+
+        section.title("Tool Details")
+        section.addFact("Version", self.tool_version)
+        section.addFact(
+            "Source code",
+            f"[{self.tool_name}]({self.source_code_url})",
+        )
+        return section
+
+    @functools.cached_property
+    def _system_info_card(self) -> "pymsteams.cardsection":
+        """Create cardsection containing PC / VM system information."""
+        assert pymsteams is not None, "module 'pymsteams' not loaded"
+        info = ctk.SystemInformation.load()
+
+        section = pymsteams.cardsection()
+        section.title("System Information")
+        section.text("Information about the PC running the scheduler.")
+
+        for name in info.__dataclass_fields__:  # pylint: disable=no-member
+            value = getattr(info, name)
+            if value is None:
+                value = "unknown"
+            elif name == "total_ram":
+                value = _common.bytes2human(value)
+            name = name.replace("_", " ").title()
+
+            for old in ("Pc", "Cpu", "Ram"):
+                name = name.replace(old, old.upper())
+
+            section.addFact(name, value)
+
+        return section
+
+    def post(
+        self, title: str, message: str, color: str = "ffffff", summary: Optional[str] = None
+    ) -> None:
+        """Post a message to the Teams channel.
+
+        Parameters
+        ----------
+        title : str
+            Title of the post.
+        message : str
+            Main body of the post, should use markdown formatting.
+        color : str,  default "ffffff" (white)
```

Comment:
Is this frame colour or text colour?

Only thinking if this is white text for teams in dark mode, would it show on teams not in dark mode